impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -779,6 +779,26 @@ public K getKey() {
     }
   }
 
+  private OnHeapValueHolder<V> invalidateInGetorComputeIfAbsent(MapWrapper<K, V> map, final K key, final ValueHolder<V> value, final Fault<V> fault, final long now) {
+    final AtomicReference<OnHeapValueHolder<V>> toInvalidate = new AtomicReference<OnHeapValueHolder<V>>();
+    map.computeIfPresent(key, new BiFunction<K, OnHeapValueHolder<V>, OnHeapValueHolder<V>>() {
+      @Override
+      public OnHeapValueHolder<V> apply(K mappedKey, final OnHeapValueHolder<V> mappedValue) {
+        if(mappedValue.equals(fault)) {
+          try {
+            toInvalidate.set(cloneValueHolder(key, value, now, Duration.ZERO, true));
+          } catch (LimitExceededException ex) {
+            throw new AssertionError("Should have already been thrown.");
```

Comment:
That message is confusing. Just say that it should not be thrown here as sizing is not happening.